sui/tokens.js

```diff
@@ -43,48 +43,23 @@ class CoinManager {
     }
 
     static async splitCoins(tx, coinTypeToCoins, options) {
-        CoinManager.checkSplitAmount(options);
-        const splitAmount = BigInt(options.split);
-
-        console.log('\n==== Splitting Coins ====');
+        printInfo('\n==== Splitting Coins ====');
 
-        // Set coin type to given coin type or the first coin type if there's only one if it's a SUI token.
-        const hasOnlyGasToken = Object.keys(coinTypeToCoins).length === 1 && coinTypeToCoins[SUI_COIN_ID];
-        const coinType = options.coinType || hasOnlyGasToken ? SUI_COIN_ID : undefined;
-
-        if (coinType) {
-            // Throw an error if the coin type is specified but no coins are found
-            CoinManager.checkCoinType(coinType, coinTypeToCoins);
-            const coins = coinTypeToCoins[coinType];
-            const [coin] = CoinManager.doSplitCoins(tx, coins, splitAmount);
+        const splitAmount = BigInt(options.split);
+        const coinType = options.coinType || SUI_COIN_ID;
 
-            if (options.transfer) {
-                CoinManager.doTransfer(tx, coin, options.transfer);
-            }
-        } else {
-            for (const coinType in coinTypeToCoins) {
-                const coins = coinTypeToCoins[coinType];
-                if (this.isGasToken(coins.data[0])) continue;
-                const [coin] = CoinManager.doSplitCoins(tx, coins, splitAmount);
-
-                if (options.transfer) {
-                    CoinManager.doTransfer(tx, coin, options.transfer);
-                }
-            }
-        }
+        const coins = coinTypeToCoins[coinType];
+        const [coin] = CoinManager.doSplitCoins(tx, coins, splitAmount);
 
         if (options.transfer) {
-            console.log(`\nTransfer ${splitAmount} coins for every split coin to ${chalk.green(options.transfer)}`);
+            tx.transferObjects([coin], options.transfer);
+            printInfo(`\nTransfer ${splitAmount} coins for every split coin to ${chalk.green(options.transfer)}`);
         }
 
         // The transaction will fail if the gas budget is not set for splitting coins transaction
         tx.setGasBudget(1e8);
     }
 
-    static doTransfer(tx, coin, recipient) {
-        tx.transferObjects([coin], recipient);
-    }
-
     static doSplitCoins(tx, coins, splitAmount) {
```

Comment:
merge this with splitCoins. There seem to be too many helpers rn. splitCoins is much smaller now